data/process_offline_test_data.py

```diff
@@ -0,0 +1,141 @@
+import json
+from datetime import datetime
+from uuid import UUID, uuid4
+
+import pandas as pd
+from tqdm import tqdm
+
+from poprox_concepts import Article, ClickHistory
+
+
+def custom_encoder(obj):
+    if isinstance(obj, UUID):
+        return str(obj)
+    elif isinstance(obj, datetime):
+        return obj.isoformat()
+
+
+def load_data():
+    val_behavior_df = pd.read_table(
+        "./val_mind_large/behaviors.tsv",
+        header="infer",
+        usecols=range(5),
+        names=["impression_id", "user", "time", "clicked_news", "impressions"],
+    )
+
+    val_behavior_df.fillna("", inplace=True)
+
+    val_news_df = pd.read_table(
+        "./val_mind_large/news.tsv", header="infer", usecols=range(4), names=["id", "topic", "subtopic", "title"]
+    )
+
+    print(
+        f"The total unique uuid is {val_behavior_df.shape[0] + val_news_df.shape[0]}"
+    )  # The total unique uuid is 448494
+    return val_behavior_df, val_news_df
+
+
+def assign_uuid(val_behavior_df, val_news_df):
+    uuids = [uuid4() for _ in range(448494)]
```

Comment:
Let's not hard-code data lengths here — obtain from the appropriate place.

We can also make this deterministic, and yeet the string / uuid distinction, and also not need the length at all:

```python
ns_article = uuid5(NAMESPACE_URL, "https://data.poprox.io/mind-large/articles/")
val_news_df["uuid"] = [str(uuid5(ns_article, aid)) for aid in val_news_df["id"]]
# `str_uuid` is no longer needed
```

Behavior UUIDs can be the same — define an appropriate namespace, and use the impression ID.